### PR TITLE
Create auto cdi feature for jakarta websockets

### LINKED ISSUE #12277: Investigate any websocket auto features that need updates for Jakarta EE9
There is at least one auto feature for CDI and websockets.

This auto feature enables the following bundle `com.ibm.ws.wsoc.cdi.weld`

We should also move the CDI tests for websockets over to open liberty.

### LINKED ISSUE #12994: Enable CDI with Websocket on websocket FAT
Porting over the CDI-Websocket test from Commercial Liberty we noticed that there were some dependencies on some jakarta features that are still under work. We decided to skip them until they are delivered

### PR DESCRIPTION
**Pull request by volosied**
fixes #12277 -- auto feature created 
fixes #12994 -- enabled skipped over skipped over tests 

if statements (as as one below) were added since those counters are static variables, and they need to be reset on the second run. 
```
        if (sequenceCounter12 == 2) {
            sequenceCounter12 = 0;
        }
```

the counter exists in the tests as the order of the unit tests is not guaranteed.  

testCdiProgrammaticEndpointCDI20 & testCdiProgrammaticEndpointCDI12 were changed to LITE so that both run in LITE & FULL modes. This ensures the check above works properly in both runs. ( Else the counter would differ in lite since only 1 test runs, and 2 in full)

### CODE REVIEW

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=io.openliberty.wsoc.internal_fat

**Comment by volosied** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+symbolicName=io.openliberty.cdi3.0-websocket2.0.feature
+visibility=private
+IBM-App-ForceRestart: install, \
+ uninstall
+IBM-Provision-Capability: \
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.websocket-2.0)))", \
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.cdi-3.0)))"
+IBM-Install-Policy: when-satisfied
+-bundles=com.ibm.ws.wsoc.cdi.weld.jakarta
+kind=noship
```
Comment (pnicolucci) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature:11]: Can you open an issue to get this put into beta once we get it delivered and into some builds? Assign it to me and I'll take care of getting it completed.

Comment (pnicolucci) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature:11]: Also in that same issue we should note looking at adding `WLP-Activation-Type: parallel`

Comment (volosied) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature:11]: Ah, that's right.

**Comment by volosied** [comment]
#build

**Review by isaacrivriv — APPROVED** [review]
Besides what Paul mentioned, looks good. Thanks for the work on this!

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+edition=core
+WLP-Activation-Type: parallel
```
Comment (pnicolucci) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature:13]: Should the edition be `full`?

Comment (pnicolucci) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-websocket2.0.feature:13]: Nvm you are correct here I was confused :)

**Review by pnicolucci — APPROVED** [review]
Changes look good now!

**Comment by pnicolucci** [comment]
#run-libby-bot

**Review by cbridgha — APPROVED** [review]
LGTM

**Pull request merged**